universal-login-relayer/src/integration/ethereum/WalletContractService.ts

```diff
@@ -1,37 +1,45 @@
-import {providers, Contract, utils} from 'ethers';
-import {calculateMessageHash, SignedMessage} from '@universal-login/commons';
-import {WalletContractInterface} from '@universal-login/contracts';
-import {getKeyFromHashAndSignature} from '../../core/utils/encodeData';
-import {beta2} from '@universal-login/contracts';
-
-export class WalletContractService {
-  constructor(private provider: providers.Provider) {
+import {Beta2Service} from './Beta2Service';
+import {BlockchainService} from '@universal-login/contracts';
+import {SignedMessage} from '@universal-login/commons';
+import IWalletContractService from '../../core/models/IWalletContractService';
+
+export class WalletContractService implements IWalletContractService {
+  constructor(private blockchainSerivce: BlockchainService, private beta2Service: Beta2Service) {
+
   }
 
-  async getRequiredSignatures(walletAddress: string): Promise<utils.BigNumber> {
-    const walletContract = new Contract(walletAddress, WalletContractInterface, this.provider);
-    const requiredSignatures = await walletContract.requiredSignatures();
-    return requiredSignatures;
+  private async getWalletService(walletAddress: string): Promise<IWalletContractService> {
+    const walletVersion = await this.blockchainSerivce.fetchWalletVersion(walletAddress);
```

Comment:
Would be good to add simple in-memory caching:
- wallet version never changes
- there will be many request to on wallet